src/js/abstract/hocs/with-shadow.js

```diff
@@ -0,0 +1,18 @@
+const withShadow = Base =>
+  class WithShadow extends Base {
+    /**
+     * Attach shadow DOM upon connect.
+     *
+     * @param {String} [mode='open']
+     */
+    connectedCallback(mode = 'open') {
```

Comment:
I didn't know that `connectedCallback` receives a `mode` arg.
Where is that specified?

Reply:
According to these shadow DOM examples, attachShadow is called in the constructor.
https://developers.google.com/web/fundamentals/web-components/shadowdom

Reply:
I cant remeber what the idea was there to be honest..

Reply:
my code comes also from a time where it was not standard yet

Reply:
okidok